client/pages/result.jsx

```diff
@@ -1,45 +1,109 @@
 import React from 'react';
 import { parseRoute } from '../../lib';
 import MapsComponent from '../components/google-maps';
-
+import Accordion from '../components/accordion';
 export default class Result extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      maps: null,
+      listOfStarsRatings: [],
       result: {
         name: '',
         location: '',
-        image: ''
+        image: '',
+        rating: null
       },
-      maps: null
+      reviews: [
+        {
+          review: {
+            user: {
+              name: ''
+            },
+            text: ''
+          }
+        },
+        {
+          review: {
+            user: {
+              name: ''
+            },
+            text: ''
+          }
+        },
+        {
+          review: {
+            user: {
+              name: ''
+            },
+            text: ''
+          }
+        }
+      ]
     };
     this.handleSearch = this.handleSearch.bind(this);
+    this.renderStars = this.renderStars.bind(this);
   }
 
   componentDidMount() {
     this.handleSearch();
   }
 
+  renderStars() {
+    const rating = this.state.result.rating;
+    const hasHalfStar = rating % 1 !== 0;
+    const fullStar = Math.floor(rating);
+    const emptyStar = Math.floor(5 - rating);
+    const arrayStars = [];
+    for (let i = 1; i <= fullStar; i++) {
+      arrayStars.push(<i key={`${i}-full-star`} className="fa-solid fa-star"></i>);
+    }
+    if (hasHalfStar) {
+      arrayStars.push(<i key="1-half-star" className="fa-solid fa-star-half-stroke"></i>);
+    }
+
+    for (let i = 1; i <= emptyStar; i++) {
+      arrayStars.push(<i key={`${i}-empty-star`} className="fa-regular fa-star"></i>);
+    }
+    this.setState({ listOfStarsRatings: arrayStars });
+  }
```

Comment:
Instead of setting these in state, return them so you can call this method in `render()`. We don't normally stick react elements in state.